cmake/onnxruntime_providers_cpu.cmake

```diff
@@ -144,6 +151,11 @@ if (HAS_BITWISE_INSTEAD_OF_LOGICAL)
   target_compile_options(onnxruntime_providers PRIVATE "-Wno-bitwise-instead-of-logical")
 endif()
 
+if(USE_NEURAL_SPEED)
+  target_link_libraries(onnxruntime_providers PRIVATE bestla::bestla)
+  set_target_properties(onnxruntime_providers PROPERTIES COMPILE_WARNING_AS_ERROR OFF) # ignore warnings inside neural-speed
```

Comment:
is this disabling warnings as errors for all source files in `onnxruntime_providers`? that may be too much.

Reply:
@edgchen1  As bestla is a header-only target, it's not meaningful to add this property to this target. To minimize the code influence, I should add a new static target for code that uses bestla's templates.

Reply:
modified in f11d0e441cbd3fba6909a8fb39ed38bd07558632

Reply:
are there plans to fix the warnings? I hope something like this is only temporary.

Reply:
@luyu-intel , you can use this way to ignore warnings from external headers:

https://github.com/microsoft/onnxruntime/blob/main/include/onnxruntime/core/common/eigen_common_wrapper.h

Here external means the files are not in the ONNX Runtime project. On the other hand, if the warning was from a file that is in the ONNX Runtime project that we can write and change,  we need to fix the warnings or suppress the warnings if the warnings are false positive.

Reply:
> are there plans to fix the warnings? I hope something like this is only temporary.

@edgchen1 Yes, we are going to fix the warnings. But some warnings which come from Xbyak (JIT header library), as a header library, can not be fixed by us.

Reply:
@snnn Thanks! I'm going to add a header wrapper for our library.

Reply:
@snnn Already changed.